packages/node_modules/@ciscospark/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -0,0 +1,520 @@
+    }
+
+    return this.request({
+      method: 'GET',
+      service: 'ediscovery',
+      resource: `reports/${reportId}`
+    })
+      .then((res) => {
+        if (res.body.reportRequest && res.body.reportRequest.encryptionKeyUrl && res.body.reportRequest.keywords) {
+          return this._decryptRequestParams(res.body.reportRequest.encryptionKeyUrl, res.body.reportRequest.name, res.body.reportRequest.description, res.body.reportRequest.keywords)
+            .then((decryptedRequestInfo) => {
+              res.body.reportRequest.name = decryptedRequestInfo.decryptedName;
+              res.body.reportRequest.description = decryptedRequestInfo.decryptedDescription;
+              res.body.reportRequest.keywords = decryptedRequestInfo.decryptedKeywords;
+              return res;
+            });
+        }
+        return res;
+      })
+      .catch((reason) => {
+        this.logger.error(`Get Report Error ${reason}`);
+        return reason;
+      });
+  },
+
+  /**
+   * Retrieves all the compliance officers reports
+   *
+   * @returns {Promise<ResponseEntity<Array<ReportRecord>>>} Http Response containing a list of report records
+   */
+  @waitForValue('@')
+  getReports() {
+    return this.request({
+      method: 'GET',
+      service: 'ediscovery',
+      resource: 'reports'
+    })
+      .then((res) => {
+        const promises = [];
+        for (let i = 0; i < res.body.length; i += 1) {
+          if (res.body[i].reportRequest && res.body[i].reportRequest.encryptionKeyUrl && res.body[i].reportRequest.keywords) {
+            promises.push(this._decryptRequestParams(res.body[i].reportRequest.encryptionKeyUrl, res.body[i].reportRequest.name, res.body[i].reportRequest.description, res.body[i].reportRequest.keywords)
+              .then((decryptedRequestInfo) => {
+                res.body[i].reportRequest.name = decryptedRequestInfo.decryptedName;
+                res.body[i].reportRequest.description = decryptedRequestInfo.decryptedDescription;
+                res.body[i].reportRequest.keywords = decryptedRequestInfo.decryptedKeywords;
+              }));
+          }
+        }
+        return Promise.all(promises)
+          .then(() => res);
+      })
+      .catch((reason) => {
+        this.logger.error(`Get Reports Error ${reason}`);
+        return reason;
+      });
+  },
+
+  /**
+   * Deletes a specified report
+   *
+   * @param {UUID} reportId - Id of the report being requested for deletion
+   * @returns {Promise<ResponseEntity>} HttpResponse indicating if delete was successful
+   */
+  @waitForValue('@')
+  deleteReport(reportId) {
+    if (!reportId) {
+      throw Error('Invalid parameter(s)');
+    }
+
+    return this.request({
+      method: 'DELETE',
+      service: 'ediscovery',
+      resource: `reports/${reportId}`
+    })
+      .then((res) => res)
+      .catch((reason) => {
+        this.logger.error(`Delete Report Error ${reason}`);
+        return reason;
```

Comment:
I think you might want to rethrow the err reason here otherwise it will fail silently

Reply:
Definitely